### DISCUSSION TITLE
本人兽医专业刚自学doris，碰到doris处理海量数据的几个问题，头大盼大佬解惑

### DISCUSSION THREAD

**Discussion by nospeak12345** [apache/doris#26582] [A - General / Q&A]
问题是这样的：
1、有20亿条数据，每次用mysql unbufferd的方式读取数据，读取前要排序，发现他会隐含一个limit65535在sql里，这个最大可以设置为多少？内存是不是会爆太大了的话？如何实现分页查询有没有经典的方法？
2、我的分区字段是taskid这个字段，是否可以修改分区字段的值？（改了后它应该去到另外的分区去了），改了后数据会不会自动往其他分区移动？
3、我发现查询这20亿条数据比较慢，是否有手动建索引或其他优化方式啊？
4、后续我要扩展我的5个BE节点为10个，扩展节点后，原来5个节点上的分区数据会自动均衡移动到后续的节点吗？

太难了！望大佬解惑，或者哪里有相应的权威教程之类的不？

**Comment by LemonLiTree** [comment]
which doris version  are you currently using

**Reply by nospeak12345** [reply]
PALO 1.1.0

**Reply by LemonLiTree** [reply]
answer your question above
1. It is not recommended to use mysql to read doris data. We have implemented the arraw protocol to read large amounts of data, which is currently in the internal testing stage. 
2. Modifying the value of a partition field is not supported
3. Your version is too low. It is recommended to upgrade to version 2.0 for faster query performance.
4. Data will be automatically balanced

**Reply by nospeak12345** [reply]
thanks！
i have to clarify the first question above，i use python3 to read datas from doris ,and i could only use mysql python api to read it,what is the better way to do it in python3?
for the 3th answer,is it safe for the datas stored  in doris if i upgrade my doris` version?i dont know how to upgrade it without the risk of lossing datas，是否有可靠的文档学习一下，thanks!

**Reply by ixzc** [reply]
It is very safe.
then, you can refer this to upgrade you doris : https://doris.apache.org/zh-CN/docs/dev/admin-manual/cluster-management/upgrade

**Comment by LemonLiTree** [comment]
https://github.com/apache/doris/issues/25514
